File: software/examples/perl/example_magnetic_flux_density.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletHallEffectV2;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your Hall Effect Bricklet 2.0

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $he = Tinkerforge::BrickletHallEffectV2->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Get current Magnetic Flux Density
my $magnetic_flux_density = $he->get_magnetic_flux_density();
print "Magnetic Flux Density: $magnetic_flux_density µT\n";

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/ruby/example_magnetic_flux_density.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_hall_effect_v2'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Hall Effect Bricklet 2.0

ipcon = IPConnection.new # Create IP connection
he = BrickletHallEffectV2.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get current Magnetic Flux Density
magnetic_flux_density = he.get_magnetic_flux_density
puts "Magnetic Flux Density: #{magnetic_flux_density} µT"

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
